Make CRE not a WARL field (#332)
riscv/riscv-cheri: src/riscv-hybrid-integration.adoc

* Mode (M)={CAP_MODE_VALUE} indicates pass:attributes,quotes[{cheri_cap_mode_name}].
* Mode (M)={INT_MODE_VALUE} indicates pass:attributes,quotes[{cheri_int_mode_name}].

The current CHERI execution mode is given by the <<m_bit>> of the <<pcc>> and the <<section_cheri_disable,CHERI register access settings>> as follows:

* The Mode is pass:attributes,quotes[{cheri_cap_mode_name}] when the <<m_bit>> of the <<pcc>> is {CAP_MODE_VALUE}, *and* <<section_cheri_disable,CHERI register access is enabled>> for the current privilege.
* Otherwise the Mode is pass:attributes,quotes[{cheri_int_mode_name}].

> When the <<m_bit>> can be set follows the rules defined by <<ACPERM>>.

Reviewer: ```suggestion
When the <<m_bit>> can be set, the rules defined by <<ACPERM>> must be followed.
```

include::img/menvcfgmodereg.edn[]

The CHERI Register Enable (CRE) bit controls whether less privileged levels can

> perform explicit accesses to CHERI registers. perform explicit accesses to CHERI registers. When the CRE bit is set to 1 and <<mseccfg>>.CRE=1,

Reviewer 2: ```suggestion
perform explicit accesses to CHERI registers. When the CRE bit is set to 1 and <<mseccfg>>.CRE=1,
```

> perform explicit accesses to CHERI registers. When CRE=1 and <<mseccfg>>.CRE=1,
> CHERI registers can be read and written by less privileged levels. When CRE=0,
> CHERI registers are disabled in less privileged levels as described in

Reviewer 3: ```suggestion
perform explicit accesses to CHERI registers. perform explicit accesses to CHERI registers. When the CRE bit is set to 1 and <<mseccfg>>.CRE=1,
CHERI registers can be read and written by less privileged levels. When set to 0,
CHERI registers are disabled in less privileged levels as described in
```

I briefly got confused here whether we are talking about the effective CRE or the CRE bit here, maybe worth being explicit?

include::img/senvcfgreg.edn[]

The CHERI Register Enable (CRE) bit controls whether U-mode can perform

> explicit accesses to CHERI registers.  When CRE=1 and <<menvcfg>>.CRE=1 and

Reviewer 3: Same comment as above.
